### DISCUSSION TITLE
Feature request: Radius Search

### DISCUSSION THREAD

**Discussion by oliverbeck-dev** [algolia/instantsearch#6085] [Q&A] [answered]
I'm looking for a Geo Radius Search Widget. I saw this example https://github.com/algolia/doc-code-samples/tree/master/vue-instantsearch/geo-search, but it doesn't really fit my requirements.

my index items have 2 fields:  the city where they belong to and the exact position lat/long. I need a widget which can filter by city and optionally add aroundLatLng/aroundRadius if needed. I don't need a map.

did someone else have a similar use case?

**Comment by Haroenv** [comment, marked as answer]
You can use Configure to pass `aroundLatLng` and `aroundRadius` based on your users input

**Reply by oliverbeck-dev** [reply]
thank you, this basically works. but I have another requirement. as soon as `aroundLatLng` and `aroundRadius` get applied, I need to remove a selected refinement from a RefimenentList. I need to do this, otherwise the final query would be wrong

**Reply by Haroenv** [reply]
In that case, a widget like this may be interesting to look at as well:

```jsx
function Center() {
  const { setIndexUiState } = useInstantSearch();

  return (
    <button
      onClick={() => {
        setIndexUiState((uiState) => ({
          ...uiState,
          configure: {
            aroundLatLng: '...',
            aroundRadius: '...',
          },
          refinementList: {},
        }));
      }}
    >
      center
    </button>
  );
}

// ...
function App() {
  return (
    <InstantSearch>
      <Configure />
      <Center />
    </InstantSearch>
  );
}
```

Alternatively, useClearRefinements can be used as well, depending on your exact use case
